java/semantickernel-api/src/main/java/com/microsoft/semantickernel/tokenizers/GPT3Tokenizer.java

```diff
@@ -0,0 +1,286 @@
+                        (char) 0x007D, (char) 0x007E, (char) 0x0121,
+                (char) 0x0122, (char) 0x0123, (char) 0x0124, (char) 0x0125, (char) 0x0126,
+                        (char) 0x0127, (char) 0x0128, (char) 0x0129,
+                (char) 0x012A, (char) 0x012B, (char) 0x012C, (char) 0x012D, (char) 0x012E,
+                        (char) 0x012F, (char) 0x0130, (char) 0x0131,
+                (char) 0x0132, (char) 0x0133, (char) 0x0134, (char) 0x0135, (char) 0x0136,
+                        (char) 0x0137, (char) 0x0138, (char) 0x0139,
+                (char) 0x013A, (char) 0x013B, (char) 0x013C, (char) 0x013D, (char) 0x013E,
+                        (char) 0x013F, (char) 0x0140, (char) 0x0141,
+                (char) 0x0142, (char) 0x00A1, (char) 0x00A2, (char) 0x00A3, (char) 0x00A4,
+                        (char) 0x00A5, (char) 0x00A6, (char) 0x00A7,
+                (char) 0x00A8, (char) 0x00A9, (char) 0x00AA, (char) 0x00AB, (char) 0x00AC,
+                        (char) 0x0143, (char) 0x00AE, (char) 0x00AF,
+                (char) 0x00B0, (char) 0x00B1, (char) 0x00B2, (char) 0x00B3, (char) 0x00B4,
+                        (char) 0x00B5, (char) 0x00B6, (char) 0x00B7,
+                (char) 0x00B8, (char) 0x00B9, (char) 0x00BA, (char) 0x00BB, (char) 0x00BC,
+                        (char) 0x00BD, (char) 0x00BE, (char) 0x00BF,
+                (char) 0x00C0, (char) 0x00C1, (char) 0x00C2, (char) 0x00C3, (char) 0x00C4,
+                        (char) 0x00C5, (char) 0x00C6, (char) 0x00C7,
+                (char) 0x00C8, (char) 0x00C9, (char) 0x00CA, (char) 0x00CB, (char) 0x00CC,
+                        (char) 0x00CD, (char) 0x00CE, (char) 0x00CF,
+                (char) 0x00D0, (char) 0x00D1, (char) 0x00D2, (char) 0x00D3, (char) 0x00D4,
+                        (char) 0x00D5, (char) 0x00D6, (char) 0x00D7,
+                (char) 0x00D8, (char) 0x00D9, (char) 0x00DA, (char) 0x00DB, (char) 0x00DC,
+                        (char) 0x00DD, (char) 0x00DE, (char) 0x00DF,
+                (char) 0x00E0, (char) 0x00E1, (char) 0x00E2, (char) 0x00E3, (char) 0x00E4,
+                        (char) 0x00E5, (char) 0x00E6, (char) 0x00E7,
+                (char) 0x00E8, (char) 0x00E9, (char) 0x00EA, (char) 0x00EB, (char) 0x00EC,
+                        (char) 0x00ED, (char) 0x00EE, (char) 0x00EF,
+                (char) 0x00F0, (char) 0x00F1, (char) 0x00F2, (char) 0x00F3, (char) 0x00F4,
+                        (char) 0x00F5, (char) 0x00F6, (char) 0x00F7,
+                (char) 0x00F8, (char) 0x00F9, (char) 0x00FA, (char) 0x00FB, (char) 0x00FC,
+                        (char) 0x00FD, (char) 0x00FE, (char) 0x00FF
+            };
+
+    private static final Pattern encodingRegex =
+            Pattern.compile(
+                    "'s|'t|'re|'ve|'m|'ll|'d| ?\\p{L}+| ?\\p{N}+|"
+                            + " ?[^\\s\\p{L}\\p{N}]+|\\s+(?!\\S)|\\s+");
+
+    private static final Map<String, List<String>> bpeCache = new HashMap<>();
```

Comment:
since this is a static, this is a bit of an uncontrolled memory leak as this map will grow over time. 

- Define a constant that can be changed via properties MAX_TOKENIZER_CACHE_SIZE
- Use LinkedHashMap and limit the size

Reply:
/spotless